File: Week02/14. 最长公共前缀.go
// 14. 最长公共前缀.go
package main

func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}

	res := strs[0]
	size := len(strs)
	for i := 1; i < size; i++ {
		j := 0
		for ; j < len(strs[i]) && j < len(res); j++ {
			if res[j] != strs[i][j] {
				break
			}
		}

		res = res[:j]
	}
	return res
}

File: Week02/14. 最长公共前缀.py
# 14. 最长公共前缀.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        shortest = min(strs, key=len)
        for i, ch in enumerate(shortest):
            for other in strs:
                if other[i] != ch:
                    return shortest[:i]
        return shortest


# 以下为自我练习
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        shortest = min(strs, key=len)
        for i, c in enumerate(shortest):
            for other in strs:
                if other[i] != c:
                    return shortest[:i]
        return shortest


def main():
    s = Solution()
    s.longestCommonPrefix(["c", "c"])


if __name__ == '__main__':
    main()
